web/crux/src/app/version/version.service.ts

```diff
@@ -205,25 +205,30 @@ export default class VersionService {
       })
 
       if (defaultVersion) {
-        const images = defaultVersion.images.map(
-          async image =>
-            await prisma.image.create({
-              select: {
-                id: true,
-              },
-              data: {
-                ...image,
-                id: undefined,
-                versionId: newVersion.id,
-                config: {
-                  create: {
-                    ...this.imageMapper.dbContainerConfigToCreateImageStatement(image.config),
-                    id: undefined,
+        const images = (
+          await Promise.all(
+            defaultVersion.images.map(async image => {
+              const newImage = await prisma.image.create({
+                select: {
+                  id: true,
+                },
+                data: {
+                  ...image,
+                  id: undefined,
+                  versionId: newVersion.id,
+                  config: {
+                    create: {
+                      ...this.imageMapper.dbContainerConfigToCreateImageStatement(image.config),
+                      id: undefined,
+                    },
                   },
                 },
-              },
+              })
+
+              return [image.id, newImage.id]
             }),
-        )
+          )
+        ).reduce((prev, current) => ({ ...prev, [current[0]]: current[1] }), {})
```

Comment:
This is a bit unreadable to me. Could you expand the function body and define some constant variable there, so we can know what's going on?

Reply:
Cleaned it up a bit.